### PR TITLE
Introduce output class using OpenMC API

### PR DESCRIPTION
**Pull request by alexvalentine94**
# Description

Introduce output class for openmc using the API. Read code version from statepoint for metadata. 

## Type of change

Please select what type of change this is.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New benchmark
    - [ ] Non-breaking change which entirely uses exisiting classes, structure etc
    - [ ] Breaking change which has implemented new/modified classes etc
- [x] New feature 
    - [x] Non-breaking change which adds functionality
    - [ ] Breaking change fix or feature that would cause existing functionality to not work as expected

# Testing

- Added test to make sure version is read from statepoint correctly with API. 

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] General testing 
    - [X] New and existing unit tests pass locally with my changes
    - [ ] Coverage is >80%

### CODE REVIEW

**Comment by alexvalentine94** [comment]
Look to improve codecov

**Comment by alexvalentine94** [comment]
> Look to improve codecov
Changes look to be covered.

File: jade/output.py [hunk 1]
```diff
@@ -72,51 +76,55 @@ def compare(self):
...
+                if file_name[-1] == "m":
+                    file1 = file_name
+                elif file_name[-1] == "o":
+                    file2 = file_name
+            elif code == "openmc":
+                if file_name.endswith('.out'):
+                    file1 = file_name
+                elif file_name.startswith("statepoint"):
+                    file2 = file_name
+
```
Comment (dodu94) [inline jade/output.py:117]: maybe it worth to add here a NotImplementedError if another code is requested

File: jade/output.py [hunk 2]
```diff
@@ -72,51 +76,55 @@ def compare(self):
...
-                mfile = file
-            elif file[-1] == "o":
-                ofile = file
-
-        if mfile is None or ofile is None:
+        file1 = None
+        file2 = None
+
+        for file_name in os.listdir(results_path):
+            if code == "mcnp":
```
Comment (dodu94) [inline jade/output.py:107]: this should be valid also for d1s

File: jade/output.py [hunk 3]
```diff
@@ -72,51 +76,55 @@ def compare(self):
...
+                elif file_name[-1] == "o":
+                    file2 = file_name
+            elif code == "openmc":
+                if file_name.endswith('.out'):
+                    file1 = file_name
+                elif file_name.startswith("statepoint"):
+                    file2 = file_name
+                
+
+        if file1 is None or (code == "mcnp" and file2 is None):
```
Comment (dodu94) [inline jade/output.py:119]: mcnp and d1s behave in the same way here

**Review by dodu94 — CHANGES_REQUESTED** [review]
Good work, just have a look at the d1s comments

**Comment by alexvalentine94** [comment]
Thanks Davide, I have implemented the changes as you suggested.

**Review by dodu94 — APPROVED** [review]

**Pull request merged**